inspire_mitmproxy/dispatcher.py

```diff
@@ -23,35 +23,55 @@
 """Dispatcher forwards requests to Services."""
 
 from logging import getLogger
-from typing import List, Type, cast
+from typing import Any, Dict, List, Optional, cast
 
 from mitmproxy.http import HTTPFlow, HTTPResponse
 
 from .errors import DoNotIntercept, NoServicesForRequest
 from .http import MITMRequest, MITMResponse
-from .services import (
-    ArxivService,
-    BaseService,
-    LegacyService,
-    ManagementService,
-    RTService,
-    WhitelistService
-)
+from .services import BaseService, ManagementService, get_service_class_from_type
 
 
 logger = getLogger(__name__)
 
 
 class Dispatcher:
-    SERVICE_LIST: List[Type[BaseService]] = [
-        ArxivService,
-        LegacyService,
-        RTService,
-        WhitelistService,
+    DEFAULT_SERVICE_LIST: List[Dict[str, Any]] = [
+        {
+            'name': 'ArxivService',
+            'type': 'BaseService',
+            'hosts_list': ['arxiv.org', 'export.arxiv.org'],
+        },
+        {
+            'name': 'LegacyService',
+            'type': 'BaseService',
+            'hosts_list': ['inspirehep.net'],
+        },
+        {
+            'name': 'RTService',
+            'type': 'BaseService',
+            'hosts_list': ['inspirevm13.cern.ch', 'rt.inspirehep.net'],
+        },
+        {
+            'name': 'WhitelistService',
+            'type': 'WhitelistService',
+            'hosts_list': ['test-indexer', 'test-scrapyd', 'test-web-e2e.local'],
+        },
     ]
 
-    def __init__(self) -> None:
-        self.services = [service_class() for service_class in self.SERVICE_LIST]
+    def __init__(self, service_list: Optional[List[Dict[str, Any]]]=None) -> None:
+        service_list = service_list or self.DEFAULT_SERVICE_LIST
+        self.set_service_list(service_list=service_list)
+
+    def set_service_list(self, service_list: List[Dict[str, Any]]):
+        self.services = [
+            get_service_class_from_type(
+                service_info.get('type', 'BaseService')
+            )(
```

Comment:
Normally I'd be at the forefront of this syntax, but being the devil's advocate, this seems like a lot going on in this comprehension. On the other hand, I tried the alternative and it doesn't look any better. Thought I'd point it out in case you have any ideas, but we can leave it

Reply:
Yep, I don't like it much either :/

Reply:
hmm... actually, I think that we should not be handling dicts here at all, it should be the manager translating from dict->Service, looking